spectral_cube/tests/test_casafuncs.py

```diff
@@ -96,6 +107,45 @@ def test_casa_read_stokes(data_advs, tmp_path):
     assert casacube.I.shape == cube.I.shape
 
 
+@pytest.mark.skipif(not casaOK, reason='CASA tests must be run in a CASA environment.')
+@pytest.mark.parametrize('filename', ('data_adv', 'data_advs', 'data_sdav',
+                                      'data_vad', 'data_vsad'),
+                         indirect=['filename'])
+def test_casa_numpyreader_read(filename, tmp_path):
+
+    cube = SpectralCube.read(filename)
+
+    make_casa_testimage(filename, tmp_path / 'casa.image')
+
+    casacube = SpectralCube.read(tmp_path / 'casa.image')
+
+    assert casacube.shape == cube.shape
+
+    arr = casa_image_array_reader(tmp_path / 'casa.image')
+
+    assert np.all(arr == casacube.unmasked_data[:])
+
+
+@pytest.mark.skipif(not casaOK, reason='CASA tests must be run in a CASA environment.')
+@pytest.mark.parametrize('shape', ((129,128,130), (513,128,128), (128,513,128),
+                                   (128,128,513), (512,64,64)),
+                         )
+def test_casa_numpyreader_read_shape(shape, tmp_path):
+
+    cube = SpectralCube.read(filename)
+
+    make_casa_testimage_of_shape(shape, tmp_path / 'casa.image')
```

Comment:
Do you have any control over the chunking? If so would be good to try different chunking options?

Reply:
no, the chunking is on-disk

Reply:
just to clarify, I mean can you create files on disk with different degrees of chunking?

Reply:
afaik, no.  CASA makes some choice for you, and I do not know of any way to control that.  It's at least not obviously accessible in any of the current interfaces.